src/js/components/search/table/ResultsColumnVisibleOption.jsx

```diff
@@ -0,0 +1,129 @@
+/**
+ * ResultsColumnVisibleOption.jsx
+ * Created by Lizzie Salita 05/31/17
+ * Drag and drop logic based on https://github.com/react-dnd/react-dnd/tree/master/examples/04%20Sortable/Simple
+ **/
+
+import React from 'react';
+
+import { findDOMNode } from 'react-dom';
+import { DragSource, DropTarget } from 'react-dnd';
+import flow from 'lodash/flow';
+import * as Icons from 'components/sharedComponents/icons/Icons';
+import ItemTypes from './ItemTypes';
+
+const propTypes = {
+    checked: React.PropTypes.string,
+    column: React.PropTypes.string,
+    label: React.PropTypes.string,
+    toggleColumnVisibility: React.PropTypes.func,
+    connectDragSource: React.PropTypes.func.isRequired,
+    connectDropTarget: React.PropTypes.func.isRequired,
+    index: React.PropTypes.number.isRequired,
+    isDragging: React.PropTypes.bool.isRequired,
+    id: React.PropTypes.any.isRequired,
+    moveColumn: React.PropTypes.func.isRequired
+};
+
+const columnSource = {
+    beginDrag(props) {
+        return {
+            id: props.id,
+            index: props.index
+        };
+    }
+};
+
+const columnTarget = {
+    hover(props, monitor, component) {
+        const dragIndex = monitor.getItem().index;
+        const hoverIndex = props.index;
+
+        // Don't replace items with themselves
+        if (dragIndex === hoverIndex) {
+            return;
+        }
+
+        // Determine rectangle on screen
+        const hoverBoundingRect = findDOMNode(component).getBoundingClientRect(); // eslint-disable-line react/no-find-dom-node
+
+        // Get vertical middle
+        const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
+
+        // Determine mouse position
+        const clientOffset = monitor.getClientOffset();
+
+        // Get pixels to the top
+        const hoverClientY = clientOffset.y - hoverBoundingRect.top;
+
+        // Only perform the move when the mouse has crossed half of the items height
+        // When dragging downwards, only move when the cursor is below 50%
+        // When dragging upwards, only move when the cursor is above 50%
+
+        // Dragging downwards
+        if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
+            return;
+        }
+
+        // Dragging upwards
+        if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
+            return;
+        }
+
+        // Perform the action
+        props.moveColumn(dragIndex, hoverIndex);
```

Comment:
Is it worth differentiating between a drag/hover event and a drop event? As currently written, the table updates (and triggers an API call) after every hover event, so dragging the first column to the end triggers a bunch of unnecessary API calls.